packages/kokkos/core/src/impl/Kokkos_Error.cpp

```diff
@@ -56,6 +56,28 @@
 
 namespace Kokkos {
 namespace Impl {
+static void (*violation_handler_pointer)(violation_info const&) = default_violation_handler;
+}
+
+void set_violation_handler(void (*new_handler)(violation_info const&)) {
+  violation_handler_pointer = new_handler;
+}
+
+namespace Impl {
+
+void default_violation_handler(violation_info const& info) {
```

Comment:
So to be safe, we should override this in Teuchos and have it call MPI_Abort()?

Reply:
Yes, if one is concerned about `std::abort` or `std::terminate` causing an MPI hang, then `MPI_Abort` should be called. Since Kokkos is determined to ignore the existence of MPI, this must be done in a handler outside of Kokkos. I personally have never seen this issue happen, but won't argue against it.